File: backend/src/main/java/com/project/service/EntityService.java
package com.project.service;

import com.project.extractor.CompanyExtractor;
import com.project.extractor.LocationExtractor;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Service
public class EntityService {

    private final CompanyExtractor companyExtractor;
    private final LocationExtractor locationExtractor;

    public EntityService() {
        this.companyExtractor = new CompanyExtractor();
        this.locationExtractor = new LocationExtractor();
    }

    public Map<String, List<String>> extractEntities(String text) {
        Map<String, List<String>> entities = new HashMap<>();
        
        entities.put("companies", companyExtractor.extract(text));
        entities.put("locations", locationExtractor.extract(text));
        
        return entities;
    }
}

File: node-backend/services/entityService.js
const COMPANIES = ["Apple", "Google", "Microsoft", "Tesla", "Amazon"];
const LOCATIONS = ["India", "USA", "China", "Germany", "Japan"];

/**
 * Extracts entities (companies and locations) from text.
 * Mimics Java's CompanyExtractor, LocationExtractor, and EntityService.
 *
 * @param {string} text
 * @returns {{companies: string[], locations: string[]}} Extracted entity lists.
 */
export function extractEntities(text) {
  const extracted = {
    companies: [],
    locations: []
  };

  if (!text) {
    return extracted;
  }

  // Find matching companies (case-insensitive with word boundary)
  for (const company of COMPANIES) {
    const regex = new RegExp(`\\b${company}\\b`, 'i');
    if (regex.test(text)) {
      extracted.companies.push(company);
    }
  }

  // Find matching locations (case-insensitive with word boundary)
  for (const location of LOCATIONS) {
    const regex = new RegExp(`\\b${location}\\b`, 'i');
    if (regex.test(text)) {
      extracted.locations.push(location);
    }
  }

  return extracted;
}
